### PR TITLE
Collision handling strategy

### PR DESCRIPTION
**Pull request by Fenn-CS**
- Added version 2 doc for collision handling strategy
- Lots of agreed changes left out intentionally, would committed as "Version 3" changes to this file for tracking purposes.

The [original V2 Google Docs document](https://docs.google.com/document/d/13zsy2amn0AeEE_cA1gkZ9vM77acY3f3nxEkA7Rvox_g/edit?usp=sharing) would/should evolve into the document for the user experience team, so diagrams and user experience questions not included here. Unless someone thinks they should?

### CODE REVIEW

File: docs/collisions.md [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+</table>
+
+
+## Deletion Risks
+
+We want to make sure that users understand the difference between a **sync **and a **copy. ** Those two commands mean different things in rclone:
+
+
+* **“sync”** is more destructive because it can delete files.  For example syncing an empty local folder with a populated Permanent archive would lead to the complete deletion of the archive which may not be what the user expected. Hence the buttons or whatever UI is exposed to the user for rclone interaction must provide the appropriate warning.
+* **“copy” **is less destructive because it does not delete files, it either creates or overwrites.  It can still cause data modification in the case of an overwrite. For example if a file exists both locally and on Permanent but with different content.  After the copy both files would contain the same content.
```
Comment (slifty) [inline docs/collisions.md:557]: typo here (space after "copy")

**Comment by slifty** [comment]
One comment that came up when I was catching up with @jasonaowen: we should explicitly note how we're going to handle characters that don't map to the filesystem (e.g. `/`).

I believe the solution would be to encode them, but let's make that explicit.

File: docs/collisions.md (hunk 1 above)
Comment (Fenn-CS) [reply docs/collisions.md]: Thanks Dan, there were a couple of those and I have fixed them.

File: docs/collisions.md [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+
+
+* A proposal was made to use the actual filesystem convention for handling name collisions which is  “space, parentheses open, number, parentheses close”  for example (2232)
+* A proposal was made to use the **recordId or FolderId **within the parenthesis instead of just natural incrementation. This would make downloadName collisions impossible and checking for available names unnecessary.
+* A proposal was made to not even store **downloadName **in the database since all the parameters that make it up are unique and already available. Instead they would be calculated as needed. [There would be a good case to store it though to make operations faster]
+* Name collisions should be checked across record and folder tables
+* [Extensions in folder names matter](https://apple.stackexchange.com/questions/123001/renamed-folder-becomes-a-file-with-an-extension) and should be handled :
+* Case sensitivity is taken into account. `Myfile` and `myFile` are different for example. That is useful to [ensure complete syncs](https://rclone.org/overview/#case-insensitive).
+
+### Reserved/unsupported characters
```
Comment (Fenn-CS) [inline docs/collisions.md:67]: @slifty commits stashed and this section added!

Comment (kfogel) [reply docs/collisions.md:67]: @Fenn-CS Guessing you meant "squashed" not "stashed"?

Comment (Fenn-CS) [reply docs/collisions.md:67]: @kfogel Yeah thanks, I mean squashed. Git and its terms 😀🤦🏽‍♂️

**Review by slifty — APPROVED** [review]
Looks great!  Thank you for this.

**Pull request merged**